lookout/style/format/benchmarks/evaluate_smoke.py

```diff
@@ -0,0 +1,195 @@
+                                    "filepath": file.path,
+                                    "style": self.config["style_name"],
+                                    "loss": file_loss})
+        self._dump_report(self.config["report_path"])
+        return []
+
+
+analyzer_class = SmokeEvalFormatAnalyser
+
+
+def report_summary(reportpath: str) -> None:
+    """
+    Print to logs report summary.
+
+    :param reportpath: path to report generated by `evaluate_smoke_entry`
+    """
+    log = logging.getLogger("report_summary")
+    no_results = 0
+    no_comments = 0
+    rest_loss = 0
+    rest_support = 0
+    cases_number = 0
+    with open(str(reportpath)) as index:
+        reader = csv.DictReader(index, fieldnames=SmokeEvalFormatAnalyser.REPORT_COLNAMES)
+        for row in reader:
+            cases_number += 1
+            loss = float(row["loss"])
+            if loss < 0:
+                no_results += 1
+            elif loss == 1:
+                no_comments += 1
+            else:
+                rest_loss += loss
+                rest_support += 1
+    log.info("Cases number: %d.", cases_number)
+    log.info("Code generation not implemented for %d cases.", no_results)
+    log.info("Failed to fix any style violation for %d cases.", no_comments)
+    log.info("Loss value for the rest cases is %.4f with support %d.",
+             rest_loss / rest_support, rest_support)
+
+
+def evaluate_smoke_entry(inputpath: str, reportpath: str) -> None:
+    """
+    CLI entry point.
+    """
+    log = logging.getLogger("evaluate_smoke")
+    port = server.find_port()
+    db = tempfile.NamedTemporaryFile(dir=inputpath)
+    fs = tempfile.TemporaryDirectory(dir=inputpath)
+    context_manager = TestAnalyzer(
+        port=port, db=db.name, fs=fs.name,
+        analyzer="lookout.style.format.benchmarks.evaluate_smoke").__enter__()
+    try:
+        inputpath = Path(inputpath)
+        if not server.file.exists():
+            server.fetch()
+        index_file = inputpath / "index.csv"
+        with open(str(index_file)) as index:
+            reader = csv.DictReader(index)
+            for row in reader:
+                repopath = inputpath / row["repo"]
+                config_json = {
+                    analyzer_class.name: {
+                        "repo_name": row["repo"],
+                        "style_name": row["style"],
+                        "report_path": reportpath
+                    }
+                }
+                server.run("push", fr=row["from"], to=row["to"], port=port,
+                           git_dir=str(repopath), )
+                server.run("review", fr=row["from"], to=row["to"], port=port,
+                           git_dir=str(repopath),
+                           config_json=json.dumps(config_json))
+                pass
+        log.info("Quality report saved to %s", reportpath)
+    finally:
+        fs.cleanup()
+        context_manager.__exit__(None, None, None)
```

Comment:
why not `with context_manager:`? It should perfectly fit

Reply:
right!